tests/format/typescript/conditional-types/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -431,3 +417,44 @@ type Unpacked<T> = T extends (infer U)[]
 
 ================================================================================
 `;
+
+exports[`parentheses.ts format 1`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+
+// #13275
+type Foo<T> = T extends ((...a: any[]) => infer R extends string) ? R : never;
+
+// #14275
+type Test<T> = T extends ((
+  token: TSESTree.Token
+) => token is infer U extends TSESTree.Token)
```

Comment:
@bradzacher Is it possible to be longer?

Reply:
I don't think it can be any deeper. I'm not 100% certain though. You could probably nest more conditionals in place of the `TSESTree.Token` - would be worth playing to see what's allowed syntactically.

Reply:
It seems possible and I don't know how to fix, I'm going to pretend I don't know. [`ac1aac1`](https://github.com/prettier/prettier/pull/14279/commits/ac1aac1146d57317cbfc4edebe2907f7fd9b238e)